viewer/packages/cad-model/src/picking/PickingHandler.ts

```diff
@@ -113,7 +116,35 @@ export class PickingHandler {
     }
   }
 
-  public async intersectCadNode(
+  private filterOutOfBoundCadNodes(cadNodes: CadNode[], input: IntersectInput) {
+    // Ensure the ray overlaps any point on the given models bounding box.
+    const raycaster = new THREE.Raycaster();
+    raycaster.setFromCamera(input.normalizedCoords, input.camera);
+    const ray = raycaster.ray;
+
+    const getWorldSpaceNodeBounds = (node: CadNode) =>
+      node.cadModelMetadata.scene.root.subtreeBoundingBox.clone().applyMatrix4(node.cadModelMetadata.modelMatrix);
+
+    // Remove all cadNodes that cannot be hit. Avoid a raycast against them
+    const candidateCadNodes = (
```

Comment:
How does this handle if the ray intersects behind the camera?

Reply:
As the ray is projected from the camera towards the normalized screen points, intersect will be always infront of camera